evm_loader/program/src/entrypoint.rs

```diff
@@ -285,19 +286,26 @@ fn process_instruction<'a>(
                 account_storage.get_caller_account().ok_or(ProgramError::InvalidArgument)?,
                 &H160::from_slice(from_addr), trx.nonce, &trx.chain_id)?;
 
-            payment::check_collateral_account(
-                program_id,
-                // WARNING Only for tests when base is random
-                operator_sol_info,
-                collateral_pool_sol_info,
-                collateral_pool_index as usize)?;
             payment::transfer_from_operator_to_collateral_pool(
+                program_id,
+                collateral_pool_index,
                 operator_sol_info,
                 collateral_pool_sol_info,
                 system_info)?;
 
+            let fee = U256::from(1000_u64) * trx.gas_price * U256::from(1_000_000_000_u64);
+            transfer_token(
+                token_transfer_accounts,
+                user_eth_info,
+                operator_eth_info,
```

Comment:
Add check that `user_eth_info` belongs to Ether signer and `operator_eth_info` belongs to Solana signer